### PR TITLE
feat(pubsub): implement quickstart guide

### LINKED ISSUE #4582: Setup a quickstart for Cloud Pub/Sub
This should be similar to the other quickstart programs, with a README, a extraction script, and with the quickstart instructions tested as part of the CI build.

### LINKED ISSUE #4648: Implement pubsub_quickstart_publisher sample
Shows how to create a publisher and publish a "Hello World!" message, e.g.

https://github.com/googleapis/python-pubsub/blob/b1bfc9e7ae540fed4a5ae81f9d895072a255a15d/samples/snippets/publisher.py#L83-L105

### PR DESCRIPTION
**Pull request by coryan**
Add quickstart program for Cloud Pub/Sub, and compile it as part of the
quickstart and install builds.

Fixes #4648 and fixes #4582 

---
This change is [Reviewable](https://reviewable.io/reviews/googleapis/google-cloud-cpp/4854)

### CODE REVIEW

File: .bazelignore [hunk 1]
```diff
@@ -5,3 +5,4 @@ cmake-build-release/
 google/cloud/bigtable/quickstart/
 google/cloud/spanner/quickstart/
 google/cloud/storage/quickstart/
+google/cloud/pubsub/quickstart/
```
Comment (devbww) [inline .bazelignore:8]: Should we keep these sorted?

File: BUILD [hunk 2]
```diff
@@ -40,3 +40,10 @@ cc_library(
         "//google/cloud/storage:storage_client",
     ],
 )
+
+cc_library(
+    name = "pubsub_client",
```
Comment (devbww) [inline BUILD:45]: Should we keep these sorted?

File: google/cloud/pubsub/quickstart/CMakeLists.txt [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+find_package(pubsub_client REQUIRED)
+
+# MSVC requires some additional code to select the correct runtime library
+if (VCPKG_TARGET_TRIPLET MATCHES "-static$")
+    set(CMAKE_MSVC_RUNTIME_LIBRARY "MultiThreaded$<$<CONFIG:Debug>:Debug>")
+else ()
+    set(CMAKE_MSVC_RUNTIME_LIBRARY "MultiThreaded$<$<CONFIG:Debug>:Debug>DLL")
+endif ()
+
+# Once the bigtable_client package is found, define new targets.
```
Comment (devbww) [inline google/cloud/pubsub/quickstart/CMakeLists.txt:34]: s/bigtable_client/pubsub_client/?

Comment (coryan) [reply google/cloud/pubsub/quickstart/CMakeLists.txt]: Thanks, fixed.

File: BUILD (hunk 2 above)
Comment (coryan) [reply BUILD]: Done.

File: .bazelignore (hunk 1 above)
Comment (coryan) [reply .bazelignore]: Done.

**Comment by coryan** [comment]
Rebased and addressed review comments.

**Pull request merged**